### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by amashutikov**

### CODE REVIEW

File: src/handleErrors.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      { message: errorMessages.missedText },
+      { message: errorMessages.notListedCase },
+    );
+  } else if (!textToConvert) {
+    errors.push({ message: errorMessages.missedText });
+  } else if (!caseName) {
+    errors.push({ message: errorMessages.missedCase });
+  } else if (!caseNames.includes(caseName)) {
+    errors.push({ message: errorMessages.notListedCase });
+  }
```
Comment (33Orange) [inline src/handleErrors.js:28]: You have a lot of unnecessary checks.
You need only 3: 
`!textToConvert`, `!caseName`, `caseName && !caseNames.includes(caseName)`

And avoid using `else` and `else if`. Create all cases in a single `if` construction

File: src/createServer.js [hunk 2]
```diff
@@ -1,3 +1,34 @@
...
+      res.end(JSON.stringify({ errors }));
+    } else {
+      const conversionResult = convertToCase(textToConvert, caseName);
+
+      conversionResult.targetCase = caseName;
+      conversionResult.originalText = textToConvert;
+
+      res.statusCode = 200;
+      res.end(JSON.stringify(conversionResult));
+    }
```
Comment (33Orange) [inline src/createServer.js:28]: Is `else` necessary here?
You have a case when you have the errors and default case.

 ```suggestion
    }
    const conversionResult = convertToCase(textToConvert, caseName);

    conversionResult.targetCase = caseName;
    conversionResult.originalText = textToConvert;

    res.statusCode = 200;
    res.end(JSON.stringify(conversionResult));
```

File: src/createServer.js [hunk 3]
```diff
@@ -1,3 +1,34 @@
...
+      res.statusMessage = 'Bad request';
+
+      res.end(JSON.stringify({ errors }));
+    } else {
+      const conversionResult = convertToCase(textToConvert, caseName);
+
+      conversionResult.targetCase = caseName;
+      conversionResult.originalText = textToConvert;
+
+      res.statusCode = 200;
```
Comment (33Orange) [inline src/createServer.js:26]: `You should respond with status 200 and OK status text.`

**Review by 33Orange — CHANGES_REQUESTED** [review]
Almost done

A few moments to fix

File: src/createServer.js [hunk 4]
```diff
@@ -1,3 +1,43 @@
...
+      res.end(JSON.stringify({ errors }));
+
+      return;
+    }
+
+    const { originalCase, convertedText } = convertToCase(
+      textToConvert, caseName,
+    );
+
+    const result = {
```
Comment (silchyn) [inline src/createServer.js:28]: That's redundant

File: src/handleErrors.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+/* eslint-disable max-len */
+const caseNames = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+const errorMessages = {
+  missedText: 'Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+  missedCase: '"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+  notListedCase: `This case is not supported. Available cases: ${caseNames.join(', ')}.`,
+};
```
Comment (silchyn) [inline src/handleErrors.js:7]: Could be in a separate file like `constants.js`

File: src/handleErrors.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+function handleErrors(textToConvert, caseName) {
```
Comment (silchyn) [inline src/handleErrors.js:9]: Do you actually handle errors here?

**Review by silchyn — APPROVED** [review]

**Pull request closed without merging**